lib/collection/src/update_handler.rs

```diff
@@ -328,7 +335,11 @@ impl UpdateHandler {
                                 stopped,
                             ) {
                                 // Perform some actions when optimization if finished
-                                Ok(result) => {
+                                Ok(num_points_optimized) => {
+                                    let mut points_indexed_in_shard = points_indexed_once.lock();
+                                    *points_indexed_in_shard += num_points_optimized;
+
+                                    let result = num_points_optimized > 0;
                                     tracker_handle.update(TrackerStatus::Done);
                                     callback(result);
                                     result
```

Comment:
Let's give `result` a name here

```suggestion
                                Ok(num_points_optimized) => {
                                    *points_indexed_once.lock() += num_points_optimized;
                                    let is_optimized = num_points_optimized > 0;
                                    tracker_handle.update(TrackerStatus::Done);
                                    callback(is_optimized);
                                    is_optimized
```